Engine/Engine.cs

```diff
@@ -236,18 +236,22 @@ public void Run(AlgorithmNodePacket job, AlgorithmManager manager, string assemb
                         initializeComplete = false;
                         //Get all the error messages: internal in algorithm and external in setup handler.
                         var errorMessage = string.Join(",", algorithm.ErrorMessages);
+                        string stackTrace = "";
                         errorMessage += string.Join(",", AlgorithmHandlers.Setup.Errors.Select(e =>
                         {
                             var message = e.Message;
                             if (e.InnerException != null)
                             {
                                 var interpreter = StackExceptionInterpreter.Instance.Value;
                                 var err = interpreter.Interpret(e.InnerException);
-                                message += interpreter.GetExceptionMessageHeader(err);
+                                var stackMessage = interpreter.GetExceptionMessageHeader(err);
+                                message += stackMessage;
+                                stackTrace += stackMessage;
                             }
                             return message;
                         }));
                         Log.Error("Engine.Run(): " + errorMessage);
+                        algorithm.SetRuntimeError(new InitializeException(errorMessage, stackTrace), "Result Handler");
```

Comment:
- `SetRuntimeError` is also logging an error, so we can probably remove `Log.Error` above to avoid the duplication
- `"Result Handler"` context isn't accurate? Should be more like Algorithm Initialization